File: Others/01. Basic Input Output/basic_IO.cpp
// Small Program to Read a user's name and greet them with Abbreviated name.

#include <iostream>
#include <cstring>
#include <fstream>
#include <conio.h>

using namespace std;

char * abbreviateName(char name[])
{
    char * abbrName;
    abbrName = new char[21];
    int j = 0;

    for (int i = 0; name[i] != 0; i++)
    {
        if (i == 0)
        {
            abbrName[j++] = name[i];
            abbrName[j++] = '.';
        }
        else if (name[i - 1] == 32)
        {
            abbrName[j++] = ' ';
            abbrName[j++] = name[i];
            abbrName[j++] = '.';
        }
    }
    abbrName[j] = 0;
    return abbrName;
}

void greetFromUserInput()
{
    char strArr[101];
    string name;

    printf("\nUser Input Mode: ");
    printf("\nPlease Enter Your Name: ");

    getline(cin, name); //Not scanf("%s", name);
    strcpy(strArr, name.c_str());

    cout << "Good Morning " << abbreviateName(strArr) << "! How are you today?";
}

void greetFromFileInput()
{
    char strArr[101];
    string name;

    ifstream input("input.txt");
    getline(input, name);
    strcpy(strArr, name.c_str());

    printf("\n\nFile Input Mode: ");
    cout << "\nGood Evening " << abbreviateName(strArr) << "! How are you today?";
}

main()
{
    greetFromUserInput();
    greetFromFileInput();
    getch();
    return 0;
}

File: Others/01. Basic Input Output/basic_IO.java
// Small Program to Read a user's name and greet them with Abbreviated name.
/*  Use as below:
    G:\Temp\Programming> javac basic_IO.java
    G:\Temp\Programming> java basic_IO
*/

import java.io.FileReader;
import java.util.Scanner;
import java.util.StringTokenizer;

public class basic_IO {
    public String abbreviateName(String p_Name) {
        String result = null;
        StringTokenizer st = new StringTokenizer(p_Name);
        while (st.hasMoreTokens()) {
            String token = st.nextToken().toString();
            if (result == null)
                result = token.charAt(0) + ".";
            else
                result += " " + token.charAt(0) + ".";
        }
        return result;
    }

    public void greetFromUserInput() {
        Scanner ins = new Scanner(System.in);
        String name = "";
        System.out.println("\nUser Input Mode: ");
        System.out.print("Please Enter Your Name: ");
        name = ins.nextLine();
        System.out.println("Good Morning " + abbreviateName(name) + "! How are you today?");
        ins.close();
    }

    public void greetFromFileInput() {
        String name = null;
        try {
            Scanner ins = new Scanner(new FileReader("input.txt"));
            name = ins.nextLine();
            ins.close();
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
        System.out.println("\nFile Input Mode: ");
        System.out.println("Good Evening " + abbreviateName(name) + "! How are you today?");
    }

    public static void main(String[] args) {
        basic_IO Ins = new basic_IO();
        Ins.greetFromUserInput();
        Ins.greetFromFileInput();
    }
}